src/hooks/useTxDetails/useTxDetails.ts

```diff
@@ -116,30 +104,31 @@ export const getTransfers = (
 export const useTxDetails = (txId: string): TxDetails => {
   const tx = useAppSelector((state: ReduxState) => selectTxById(state, txId))
   const assets = useAppSelector(selectAssets)
-  const marketDataUserCurrency = useAppSelector(selectMarketDataUserCurrency)
 
-  const transfers = useMemo(
-    () => getTransfers(tx, assets, marketDataUserCurrency),
-    [tx, assets, marketDataUserCurrency],
-  )
+  const transfers = useMemo(() => {
+    if (!tx) return []
+    return getTransfers(tx, assets)
+  }, [tx, assets])
 
   const fee = useMemo(() => {
-    if (!tx.fee) return
+    if (!tx?.fee) return
     return {
       ...tx.fee,
       asset: assets[tx.fee.assetId] ?? defaultAsset,
-      marketData: marketDataUserCurrency[tx.fee.assetId] ?? defaultMarketData,
     }
-  }, [tx.fee, assets, marketDataUserCurrency])
+  }, [tx?.fee, assets])
 
-  const feeAsset = useAppSelector(state => selectFeeAssetByChainId(state, tx.chainId))
+  const feeAsset = useAppSelector(state => selectFeeAssetByChainId(state, tx?.chainId ?? ''))
 
   const txLink = getTxLink({
-    name: tx.trade?.dexName,
+    name: tx?.trade?.dexName,
     defaultExplorerBaseUrl: feeAsset?.explorerTxLink ?? '',
-    txId: tx.txid,
+    txId,
   })
 
+  // This should never happen, but if it does we should not continue
+  if (!tx) throw Error('Transaction not found')
```

Comment:
This is intended to avoid refactoring a large amount of code to handle tx being optional, which realistically shouldn't be necessary since this was already in prod without our knowledge. This line just makes it clear.